components/style/values/specified/grid.rs

```diff
@@ -223,24 +224,25 @@ impl<L> Default for TrackSize<L> {
 impl Parse for TrackSize<LengthOrPercentage> {
     fn parse(context: &ParserContext, input: &mut Parser) -> Result<Self, ()> {
         if let Ok(b) = input.try(|i| TrackBreadth::parse(context, i)) {
-            Ok(TrackSize::Breadth(b))
-        } else {
-            if input.try(|i| i.expect_function_matching("minmax")).is_ok() {
-                Ok(try!(input.parse_nested_block(|input| {
-                    let inflexible_breadth = if let Ok(lop) = input.try(LengthOrPercentage::parse_non_negative) {
-                        Ok(TrackBreadth::Breadth(lop))
-                    } else {
-                        TrackKeyword::parse(input).map(TrackBreadth::Keyword)
+            return Ok(TrackSize::Breadth(b))
+        }
+
+        if input.try(|i| i.expect_function_matching("minmax")).is_ok() {
+            return input.parse_nested_block(|input| {
+                let inflexible_breadth =
+                    match input.try(LengthOrPercentage::parse_non_negative) {
+                        Ok(lop) => TrackBreadth::Breadth(lop),
+                        Err(..) => TrackBreadth::Keyword(try!(TrackKeyword::parse(input))),
                     };
 
-                    try!(input.expect_comma());
-                    Ok(TrackSize::MinMax(try!(inflexible_breadth), try!(TrackBreadth::parse(context, input))))
-                })))
-            } else {
-                try!(input.expect_function_matching("fit-content"));
-                Ok(try!(LengthOrPercentage::parse(context, input).map(TrackSize::FitContent)))
-            }
+                try!(input.expect_comma());
+                Ok(TrackSize::MinMax(inflexible_breadth, try!(TrackBreadth::parse(context, input))))
+            });
         }
+
+        try!(input.expect_function_matching("fit-content"));
+        // FIXME(emilio): This needs a parse_nested_block, doesn't it?
```

Comment:
Given that hasn't merged yet, and this needs a rebase anyway, I'd rather leave it there until that merges (btw, let me know when is it ready for re-review! I just saw a bunch of comments addressed that I wouldn't have noticed otherwise).